Feature/scrum 39 back form

cesarj45 opened the pull request:
## What kind of PR is this?

- [x ] Feature

## Description
adds user entity
adds role entity
implements new eslint rules

## Ticket id
https://barbershop-app.atlassian.net/jira/software/projects/SCRUM/boards/1?selectedIssue=SCRUM-39
{}

## Useful resources

![form entities](https://github.com/user-attachments/assets/59429e2f-89e8-4f1f-aea0-7500c29aa9fb)

LeonelCeballos18 committed:
feature: add back calendar entities

adds appointment entity
adds appointment service entity
adds customer entity
adds service entity
adds relations between entities
adds migrations
adds all entities in data source

cesarj45 committed:
feature: adds entities related with the form

cesarj45 committed:
feature: adds entities related with the form

cmagallon1 commented on src/entity/user.entity.ts:
```diff
@@ -0,0 +1,23 @@
+  Name: string;
+
+  @Column()
+  phone: number;
```
phone numbers should be a `string` data type since that makes it easier to work with rather than being numbers

cesarj45 replied:
I suspected that, thank you, I'll change it

cmagallon1 commented on src/data-source.ts:
```diff
@@ -1,5 +1,7 @@
 import 'reflect-metadata';
 import { DataSource } from 'typeorm';
+import { User } from './entity/user.entity';
```
same comment mentioned here: https://github.com/cmoreno357/mindticas2025-back/pull/7/files#r1949719055

👀

cesarj45 replied:
ack

cmagallon1 commented on src/migration/1739228224641-adds-user-and-role-entities.ts:
```diff
@@ -0,0 +1,18 @@
+export class AddsUserAndRoleEntities1739228224641 implements MigrationInterface {
+    name = 'AddsUserAndRoleEntities1739228224641'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
```
just curious, are these migrations generated automatically by typeorm or are you defining the code inside? 🤔

cesarj45 replied:
Automatically generated with the command  `yarn typeorm migration:generate ./src/migration/adds-user-and-role-entities -d src/data-source.ts`

LeonelCeballos18 committed:
chore: Solving comments

change `total_price`field from appointments from int to float and `price` field from services from int to float
change relation customers and appointments one to one
adds index file which containts all the clases

cesarj45 committed:
chore: Solving comments
- change `phone` field from number to string
- change relationship from `User` to `Role` to OneToOne
- adds `User` and `Role` paths to index.ts
- Remove the AppointmentService entity because it should be created by the TypeORM according to the documentation adding the @ManyToMany relationship
- adds bidirectional relationship among all entities following the documentation
- run lint command to fix formatting

cmoreno357 commented on tsconfig.json:
```diff
@@ -10,7 +10,6 @@
       "experimentalDecorators": true,
-      "sourceMap": true,
-   },
-      "include": ["src", "types/env.d.ts"],
```
Why tho?

cmoreno357 committed:
Merge branch 'development' into feature/SCRUM-39-back-form

plus added .DS_Store to the .gitignore file

mtopete1996 commented on src/data-source.ts:
```diff
@@ -1,14 +1,10 @@
-  Customer,
-  Service,
-} from './entity/index';
+import { Appointment, Customer, Service, User, Role } from './entity';
```
❤️

mtopete1996 commented on src/data-source.ts:
```diff
@@ -1,14 +1,10 @@
 // eslint-disable-next-line @typescript-eslint/no-var-requires
-require('dotenv').config();
+import 'dotenv/config';
+// require('dotenv').config();
```
Do we need this comment?

mtopete1996 commented on src/data-source.ts:
```diff
@@ -25,7 +21,7 @@ export const AppDataSource = new DataSource({
-  entities: [Appointment, AppointmentService, Customer, Service],
-  migrations: ['src/migrations/*.ts'],
+  entities: [Appointment, Customer, Service, User, Role],
+  migrations: ['. /migration/*.ts'],
```
typo?

mtopete1996 reviewed (commented):
I have my doubts

cmoreno357 committed:
chore: implements new eslint rules

mtopete1996 reviewed (approved):
LGTM :)
